src/evm/protocol/filters.rs

```diff
@@ -123,3 +123,22 @@ pub fn curve_pool_filter(component: &ComponentWithState) -> bool {
     }
     true
 }
+
+/// Filters out pool that have hooks in Uniswap V4
+pub fn uniswap_v4_pool_with_hook_filter(component: &ComponentWithState) -> bool {
+    if let Some(hooks) = component
+        .component
+        .static_attributes
+        .get("hooks")
+    {
+        if !hooks.to_vec().iter().all(|&b| b == 0) {
```

Comment:
Comparing to a vec made from the address 0 would be cleaner IMO, but not the biggest deal. Pls just add a comment maybe saying the address 0 == no hook